### PR TITLE
Update viewAs to use participant and instructor as view types

### PR DESCRIPTION
**Pull request by dmcavoy**
Updates the viewAsRedux to use the language of Instructor and Participant instead of Teacher and Student. (This updates the URL for viewing as teacher and student but does not update any of the features on the page.  It does not check the new participant and instructor audiences fields yet.)

In viewAsRedux, added some logic to check if we got one of the old ViewTypes (Student or Teacher) and redirect to the new ViewTypes(Instructor or Participant).

Screen Shot 2021-11-09 at 10 11 28 AM

Update the language in tests that this change impacted as well. 

## Links

- [Eng Plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#heading=h.fk6t1p96qqdl)
- [Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#heading=h.qv29hwce61yz)
- [Jira](https://codedotorg.atlassian.net/browse/PLAT-1343)

## Testing story

- Updates existing tests
- Added tests for the new redirection from old ViewTypes to new ones

### CODE REVIEW

File: apps/src/code-studio/viewAsRedux.js
Comment (davidsbailey) [inline apps/src/code-studio/viewAsRedux.js]: good call to handle these cases. will this cause the url to get updated?

Comment (dmcavoy) [reply apps/src/code-studio/viewAsRedux.js]: It does now :)

**Comment by jamescodeorg** [comment]
I hate to ask, but is this something we should wait until post-HoC to merge?  (I haven't made it through the whole review yet but it seems like a big-ish change?)

**Comment by dmcavoy** [comment]
> I hate to ask, but is this something we should wait until post-HoC to merge? (I haven't made it through the whole review yet but it seems like a big-ish change?)

I get the concern. I feel pretty comfortable with the change. It is mostly a name change and it still handles the old names so that feels like its really pretty safe. Plus if we make the change this week we still have some time before lockdown if something came up.

**Comment by davidsbailey** [comment]
Though Hour of Code lockdown date is getting close, I think this change seems reasonable to do, provided we can get it done in the next few days. My bigger concern is that, in the case of a problem shortly after deploy, this looks difficult to roll back because:

* generally, this PR changes URLs, which is a warning sign that reverting could be tricky as teachers could get new urls stuck in their autocomplete
* specifically, reverting could lead to this badly broken user experience:
Screen Shot 2021-11-10 at 9 00 39 PM
* the PR touches a large number of lines across the codebase, and a rollback is likely to have a fair number of merge conflicts

two thoughts on how to proceed here:
1. minimize severity of a rollback via an initial PR which gets merged before the PR that changes the user experience. for example, https://studio.code.org/s/csd1-2021/lessons/5/levels/1?section_id=3541304&user_id&viewAs=Participant should at least render the page and allow the teacher to get where they want by clicking more buttons in the teacher panel. Then, check for other scenarios which could be broken. For extra credit, make it so that viewAs=Participant actually gives you the same behavior as viewAs=Student. 
2. try and break this PR up a bit, so that the PR which actually makes the user-visible change is small (easy to revert), and the rest of the code change comes either before or after. this is a best-effort situation -- ultimately we can sort through some merge conflicts if needed, but doing some splitting is probably worth it here.
 
Sorry to slow the pace of progress here. I'd be happy to pair with you on this tomorrow.

**Comment by dmcavoy** [comment]
@jamescodeorg and @davidsbailey - I think with the two PRs thats you all reviewed this morning this PR is ready to be reviewed and considered for merging this week. It is now really just a find and replace on ViewType.Student and ViewType.Teacher to ViewType.Participant and ViewType.Instructor. Beyond that it is updates to comments and test names. Please give it a look and see what you think

**Review by davidsbailey — APPROVED** [review]
> @jamescodeorg and @davidsbailey - I think with the two PRs thats you all reviewed this morning this PR is ready to be reviewed and considered for merging this week. It is now really just a find and replace on ViewType.Student and ViewType.Teacher to ViewType.Participant and ViewType.Instructor. Beyond that it is updates to comments and test names. Please give it a look and see what you think

Nice job Dani! This seems like a great way to split the PRs up. The plan sounds good to me. Ideally, we would have at least one deploy between the previous two PRs and this PR reaching production. If the other two PRs can ship today, maybe we can plan to ship this tomorrow? (i.e. merge this after today's DTP)

One other nit -- this part of the initial PR description is throwing me off:

> This is purely a language update and should not impact any functionality. Does not check the new participant and instructor audiences fields yet.

This was hard for me to process because I would count changing any change to urls as a legitimate change in functionality, especially since you're also changing some logic for rewriting one set of viewAs states to the other. Would you mind updating? I think you could say that this affects URLs but does not affect any other part of the user experience.

**Comment by dmcavoy** [comment]
@davidsbailey Thanks for the feedback on the language. I have updated. 

I will wait until the other changes have got through the deploy to merge this change. Hopefully that means this can go out tomorrow

**Review by jamescodeorg — APPROVED** [review]

**Comment by jamescodeorg** [comment]
That's a lot of places where we switch on ViewType!  Nice work tracking all of these down.

**Pull request merged**
